lib/docker-toolbelt.coffee

```diff
@@ -79,9 +93,66 @@ Docker::imageRootDir = (image) ->
 					imageInfo.GraphDriver.Data.RootDir
 				when 'vfs'
 					path.join(dkroot, 'vfs/dir', destId)
+				when 'aufs'
+					path.join(dkroot, 'aufs/diff', destId)
 				else
 					throw new Error("Unsupported driver: #{dockerInfo.Driver}/")
 
+# Same as imageRootDir, but provides the full mounted rootfs for AUFS,
+# and has a disposer to unmount.
+Docker::imageRootDirMounted = (image) ->
+	docker = this
+	driver = null
```

Comment:
You're right, in that case I think it would be better to push the `disposer` only around the bit that actually needs disposing, eg
```coffee
docker.aufsDiffPaths(image)
.then (layerDiffPaths) ->
	branchesOption = 'br=' + layerDiffPaths.join('=ro:') + '=ro'
	rootDir = path.join(dkroot, 'aufs/mnt', 'tmp' + imageId.split(':')[1])
	fs.mkdirAsync(rootDir)
	.then ->
		execAsync("mount -t aufs -o 'noxino,ro,#{branchesOption}' none #{rootDir}")
	.return(rootDir)
	.disposer (rootDir) ->
		execAsync("umount #{rootDir}")
		.then ->
			fs.rmdirAsync(rootDir)
		.catch (err) ->
			# We don't want to crash the node process if something failed here...
			console.error('Failed to clean up after imageRootDirMounted', err, err.stack)
			return